File: fetch_publications.php
<?php
// filepath: c:\Users\skall\OneDrive - Chalmers\Documents\kristianssonlabofwonders\fetch_publications.php

// Fetch the Google Scholar page
$url = 'https://scholar.google.se/citations?user=WauDzPUAAAAJ&hl=en&oi=ao';
$html = file_get_contents($url);

// Use a library like simple_html_dom to parse the HTML
include('simple_html_dom.php');
$dom = str_get_html($html);

// Extract publication data
$publications = [];
foreach ($dom->find('.gsc_a_tr') as $row) {
    $title = $row->find('.gsc_a_at', 0)->plaintext;
    $link = 'https://scholar.google.se' . $row->find('.gsc_a_at', 0)->href;
    $authors = $row->find('.gsc_a_at', 0)->next_sibling()->plaintext;
    $publications[] = [
        'title' => $title,
        'link' => $link,
        'authors' => $authors
    ];
}

// Return the data as JSON
header('Content-Type: application/json');
echo json_encode($publications);
?>

File: fetch_publications.py
import requests
from bs4 import BeautifulSoup
import json

# URL of the Google Scholar profile
SCHOLAR_URL = "https://scholar.google.se/citations?hl=sv&user=WauDzPUAAAAJ&view_op=list_works&sortby=pubdate"

# Fetch the webpage content
response = requests.get(SCHOLAR_URL, headers={"User-Agent": "Mozilla/5.0"})
html = response.text

# Parse the HTML
soup = BeautifulSoup(html, "html.parser")

# Extract publication data
publications = []
for row in soup.select(".gsc_a_tr"):  # Selects each publication row
    title_element = row.select_one(".gsc_a_at")
    title = title_element.text.strip()
    link = "https://scholar.google.se" + title_element["href"]

    author_element = title_element.find_next_sibling()
    authors = author_element.text.strip() if author_element else "Unknown"

    year_element = row.select_one(".gsc_a_y")
    year = year_element.text.strip() if year_element else "Unknown"

    publications.append({
        "title": title,
        "link": link,
        "authors": authors,
        "date": year
    })

# Save the data to a JSON file
with open("publications.json", "w", encoding="utf-8") as f:
    json.dump(publications, f, ensure_ascii=False, indent=4)

print("✅ Publications JSON file updated successfully.")
